grpc/server_test.go

```diff
@@ -25,19 +26,21 @@ func Test_FetchAuthDataSync(t *testing.T) {
 					"endpoint_1": {
 						EndpointId: "endpoint_1",
 						Auth: &proto.Auth{
-							AuthType: proto.Auth_API_KEY_AUTH,
-							AuthTypeDetails: &proto.Auth_ApiKey{
-								ApiKey: "secret_key_1",
+							AuthType: proto.Auth_AUTH_TYPE_API_KEY,
+							AuthTypeDetails: &proto.Auth_StaticApiKey{
+								StaticApiKey: &proto.StaticAPIKey{
+									ApiKey: "secret_key_1",
+								},
 							},
 						},
 						RateLimiting: &proto.RateLimiting{
 							ThroughputLimit:     100,
 							CapacityLimit:       1000,
 							CapacityLimitPeriod: proto.CapacityLimitPeriod_CAPACITY_LIMIT_PERIOD_DAILY,
 						},
-						Metadata: map[string]string{
-							"account_id": "account_1",
-							"plan_type":  "PLAN_FREE",
+						Metadata: &proto.Metadata{
+							AccountId: "account_1",
```

Comment:
Having typed fields in the Metadata is a great improvement.
These should be grouped under a specific type here though, e.g. `LegayAuthDetails`, to avoid cluttering the metadata.

Reply:
I strongly feel we should not have any reference to "`Legacy`" or Grove-specific fields in the hardcoded field names. 

I also don't feel that "`AuthDetails`" correct captures the intent of this field as these details are not related to Auth at all I (they are metadata, not required for any kind of auth).

The intent of including these fields is not simply that we may need to use them but also that it captures common fields that users of PATH may need/want to use so I don't believe it is necessary to name them as suggested in your comment.

Reply:
+1 to avoid the "legacy" section. 

I'd rather just use this PR to rename/remove what's needed and what's not and I think the `Metadata` field is "metadata-ish" enough without any explicit references to Portal.

@adshmh In this specific instance, I'd rather even avoid a `TODO_TECHDEBT`.

If there is any particular field we think is unnecessary, let's just remove it. That said, from my perspective, it's 🆗